### PR TITLE
Added MftEntry::from_buffer_skip_fixup()

### PR DESCRIPTION
**Pull request by forensicmatt**
I added the function `MftEntry::from_buffer_skip_fixup()` to get a MftEntry from a buffer that already has fixup values applied. It is rare that this is a scenario, but the scenario does exist. For example, I can use the Windows API to get an MFT Entry buffer via this DeviceIoControl (https://docs.microsoft.com/en-us/windows/win32/api/winioctl/ni-winioctl-fsctl_get_ntfs_file_record). The return buffer has the fixup already applied, thus, using `MftEntry::from_buffer()` fails because of the fixup validation. However, by skipping the fixup, I can use MftEntry with these buffers. I am going to use this in my USN monitor to enumerate full paths for live monitoring.

### CODE REVIEW

File: src/entry.rs [hunk 1]
```diff
@@ -187,6 +187,29 @@ impl MftEntry {
         })
     }
 
+    /// Initializes an MFT Entry from a buffer but skips applying fixups
+    /// It is not recommended to use this function unless you know what you are doing.
```
Comment (omerbenamram) [inline src/entry.rs]: This line isn't very helpful.

We can just state what is the purpose of this API.
Also document when errors can still occur in this situation (buffer signature being wrong).

File: src/entry.rs [hunk 2]
```diff
@@ -187,6 +187,29 @@ impl MftEntry {
...
+    /// The main purpose of it is for use when you have buffers that already have fixup
+    /// already applied. For example, using Windows API 
+    /// (https://docs.microsoft.com/en-us/windows/win32/api/winioctl/ni-winioctl-fsctl_get_ntfs_file_record)
+    pub fn from_buffer_skip_fixup(buffer: Vec<u8>, entry_number: u64) -> Result<MftEntry> {
+        let mut cursor = Cursor::new(&buffer);
+        // Get Header
+        let entry_header = EntryHeader::from_reader(&mut cursor, entry_number)?;
+        trace!("Number of sectors: {:#?}", entry_header);
+
+        if !entry_header.is_valid() {
```
Comment (omerbenamram) [inline src/entry.rs]: use the `ensure!` macro here (like line 101):

```
ensure!(
            entry_header.is_valid(),
            err::InvalidEntrySignature {
                bad_sig: entry_header.signature.to_vec()
            }
        );
```

File: src/entry.rs (hunk 1 above)
Comment (forensicmatt) [reply src/entry.rs]: Added more appropriate doc message.

File: src/entry.rs (hunk 2 above)
Comment (forensicmatt) [reply src/entry.rs]: Switched to using ensure! for error handling.

**Comment by forensicmatt** [comment]
any chance you could make a new release so this change is available via cargo?

**Pull request merged**